scripts/sonic_sku_create.py

```diff
@@ -0,0 +1,805 @@
+#! /usr/bin/env python
+"""
+usage: sonic_sku_create.py [-h] [-v] [-f FILE] [-m [MINIGRAPH_FILE]] [-b BASE]
+                           [-r] [-k HWSKU]
+                           [-p] [-vv]
+Create a new SKU
+
+optional arguments:
+  -h, --help            Show this help message and exit
+  -v, --version         show program's version number and exit
+  -f FILE, --file FILE  SKU definition from xml file. -f OR -m must be provided when creating a new SKU
+  -m [MINIGRAPH_FILE], --minigraph_file [MINIGRAPH_FILE]
+                        SKU definition from minigraph file. -f OR -m must be provided when creating a new SKU
+  -b BASE, --base BASE  SKU base definition
+  -r, --remove          Remove SKU folder
+  -k HWSKU, --hwsku HWSKU
+                        SKU name to be used when creating a new SKU or for  L2 configuration mode
+  -p, --print           Print port_config.ini without creating a new SKU
+  -vv, --verbose        Verbose output
+
+
+"""
+
+from __future__ import print_function
+
+import argparse
+import json
+import os
+import re
+import subprocess
+import traceback
+import sys
+import shutil
+import copy
+from collections import OrderedDict
+
+from tabulate import tabulate
+from lxml import etree as ET
+from lxml.etree import QName
+
+minigraph_ns = "Microsoft.Search.Autopilot.Evolution"
+minigraph_ns1 = "http://schemas.datacontract.org/2004/07/Microsoft.Search.Autopilot.Evolution"
+INTERFACE_KEY = "Ethernet"
+
+test_envir = None
+try:
+    test_envir = os.environ.get('UTILITIES_UNIT_TESTING')
+except KeyError:
+    print("Not Runing on unit test envir", file=sys.stderr)
+
+### port_config.ini header
+PORTCONFIG_HEADER = ["# name", "lanes", "alias", "index", "speed"]
+platform_4 = ['x86_64-mlnx_lssn2700-r0','x86_64-mlnx_msn2010-r0','x86_64-mlnx_msn2100-r0','x86_64-mlnx_msn2410-r0','x86_64-mlnx_msn2700-r0','x86_64-mlnx_msn2740-r0','x86_64-mlnx_msn3700c-r0','x86_64-mlnx_msn3700-r0','x86_64-mlnx_msn3800-r0']
+platform_8 = ['x86_64-mlnx_msn4600c-r0','x86_64-mlnx_msn4700-r0']
```

Comment:
Why we have mlnx platforms? I thought this script is general to any platform.

Reply:
This script as of now is for creating SKU for Mellanox Platforms. To make SKUs for other platforms, we may have to bring in the details of other platforms and process specific to that platform, because, port name, lane number and other restrictions may change according to respective platform.

Reply:
Parsing of xml file or minigraph file for SKU definition does not have any Mellanox related definitions.